test/equiv.lean

```diff
@@ -0,0 +1,15 @@
+import data.set.finite
+import data.finset.basic
+
+def s : finset (fin 3) := {0, 1}
+
+example : finset (fin 3) := s.map (equiv.add_left (1 : fin 3)).to_embedding
+example : finset (fin 3) := s.map (equiv.add_left (1 : fin 3))
+
+def f : equiv (fin 3) (fin 3) :=
+{ to_fun := λ x, x + 1,
+  inv_fun := λ x, x - 1,
+  left_inv := dec_trivial,
+  right_inv := dec_trivial }
+
+example : finset (fin 3) := s.map f
```

Comment:
It's interesting to note that without the type hint, this example errors.
```lean
example := s.map f -- type mismatch at application, `f` has type `fin 3 ≃ fin 3` but is expected to have type `fin 3 ↪ ?m_1 : Type ?`
```
In particular, this causes an interesting asymmetry due to elaboration order shenanigans:
```lean
example : s.map f.to_embedding = s.map f := by simp -- works
example : s.map f = s.map f.to_embedding := by simp -- type error on `s.map f`
```

IMO that's a sufficient argument for keeping `to_embedding` as the `simp` normal form, which is worth a comment above.